src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -225,6 +226,61 @@ private byte[] CompressToken(string token, string algorithm)
             return compressionProvider.Compress(Encoding.UTF8.GetBytes(token)) ?? throw LogHelper.LogExceptionMessage(new InvalidOperationException(LogHelper.FormatInvariant(TokenLogMessages.IDX10680, algorithm)));
         }
 
+        /// <summary>
+        /// Creates a <see cref="ClaimsIdentity"/> from a <see cref="JsonWebToken"/>.
+        /// </summary>
+        /// <param name="jwtToken">The <see cref="JsonWebToken"/> to use as a <see cref="Claim"/> source.</param>
+        /// <param name="validationParameters"> Contains parameters for validating the token.</param>
+        /// <returns>A <see cref="ClaimsIdentity"/> containing the <see cref="JsonWebToken.Claims"/>.</returns>
+        protected virtual ClaimsIdentity CreateClaimsIdentity(JsonWebToken jwtToken, TokenValidationParameters validationParameters)
+        {
+            if (jwtToken == null)
+                throw LogHelper.LogArgumentNullException(nameof(jwtToken));
+
+            if (validationParameters == null)
+                throw LogHelper.LogArgumentNullException(nameof(validationParameters));
+
+            var actualIssuer = jwtToken.Issuer;
+            if (string.IsNullOrWhiteSpace(actualIssuer))
+            {
+                LogHelper.LogVerbose(TokenLogMessages.IDX10244, ClaimsIdentity.DefaultIssuer);
+                actualIssuer = ClaimsIdentity.DefaultIssuer;
+            }
+
+            return CreateClaimsIdentity(jwtToken, actualIssuer, validationParameters);
+        }
+
+        private ClaimsIdentity CreateClaimsIdentity(JsonWebToken jwtToken, string actualIssuer, TokenValidationParameters validationParameters)
```

Comment:
nit: consider moving the actualIssuer param last (will better match the signature addition to the protected virtual above)

Reply:
Done.